src/e131.c

```diff
@@ -164,6 +164,30 @@ int e131_multicast_join_iface(int sockfd, const uint16_t universe, const int ifi
   return setsockopt(sockfd, IPPROTO_IP, IP_ADD_MEMBERSHIP, &mreq, sizeof mreq);
 }
 
+/* Join a socket file descriptor to an E1.31 multicast group using a universe and an IP address to bind to */
+extern int e131_multicast_join_ifaddr(int sockfd, const uint16_t universe, const char *ifaddr) {
+  if (universe < 1 || universe > 63999) {
+    errno = EINVAL;
+    return -1;
+  }
+#ifdef _WIN32
+  if (ifindex != 0) {
+    errno = ENOSYS;
+    return -1;
+  }
+  struct ip_mreq mreq;
+  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
+  mreq.imr_interface.s_addr = inet_addr(ifaddr);
+  mreq.imr_ifindex = 0;
+#else
+  struct ip_mreqn mreq;
+  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
+  mreq.imr_address.s_addr = inet_addr(ifaddr);
+  mreq.imr_ifindex = 0;
+#endif
```

Comment:
The `_WIN32` codepath is actually incorrect as the `ifindex` variable does not exist in this function.
If we remove the unnecessary check, we end up with two codepaths that are exactly the same:
```c
#ifdef _WIN32
  struct ip_mreq mreq;
  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
  mreq.imr_interface.s_addr = inet_addr(ifaddr);
  mreq.imr_ifindex = 0;
#else
  struct ip_mreqn mreq;
  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
  mreq.imr_address.s_addr = inet_addr(ifaddr);
  mreq.imr_ifindex = 0;
#endif
```

In which case, the branching is actually redundant. Therefore, can you remove the `#ifdef` branching altogether and only leave one case? This means your code is platform independent :)

Reply:
After reading the [documentation for Windows](https://learn.microsoft.com/en-us/windows/win32/api/ws2ipdef/ns-ws2ipdef-ip_mreq) a bit now, I also noticed that `mreq.imr_ifindex` does not exist.

So in summary the final code should be:
```c
#ifdef _WIN32
  struct ip_mreq mreq;
  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
  mreq.imr_interface.s_addr = inet_addr(ifaddr);
#else
  struct ip_mreqn mreq;
  mreq.imr_multiaddr.s_addr = htonl(0xefff0000 | universe);
  mreq.imr_address.s_addr = inet_addr(ifaddr);
  mreq.imr_ifindex = 0;
#endif
```

Sorry about all the fuzz! :)

Reply:
Oh no worries! I'll fix it after work. I don't keep personal code on my work machine. 😅

Reply:
Hehe me neither! Good practice! And no rush, take your time!

Reply:
Thanks for this library, by the way! I bought a copy of the spec from ANSI and was about to start making my own when I decided to look and see if I could find one I liked that someone else had written first.

I checked it against the spec and ✅ it all matched up exactly. (And you wrote it exactly like how I would have!)

You saved me a few weeks of time, and I'm grateful.

Reply:
Removed `mreq.imr_ifindex` from the Windows code path!

Reply:
> Thanks for this library, by the way! I bought a copy of the spec from ANSI and was about to start making my own when I decided to look and see if I could find one I liked that someone else had written first.

I indeed based this library by closely reading the spec and implementing it as close and natural as possible!

I know there is a newer spec nowadays for E1.31 (sACN), which I would like to implement at some point too :).